src/sentry/static/sentry/app/components/smartSearchBar/index.tsx

```diff
@@ -503,46 +603,47 @@ class SmartSearchBar extends React.Component {
    * with data when ready
    */
   getTagValues = debounce(
-    async (tag, query) => {
+    async (tag: Tag, query: string) => {
       // Strip double quotes if there are any
       query = query.replace(/"/g, '').trim();
 
-      this.setState({
-        loading: true,
-      });
+      if (!this.props.onGetTagValues) {
+        return [];
+      }
 
-      try {
-        const {location} = this.context.router;
-        const endpointParams = getParams(location.query);
+      if (
+        this.state.noValueQuery !== undefined &&
+        query.startsWith(this.state.noValueQuery)
+      ) {
+        return [];
+      }
 
-        if (
-          this.state.noValueQuery === undefined ||
-          !query.startsWith(this.state.noValueQuery)
-        ) {
-          const values = await this.props.onGetTagValues(tag, query, endpointParams);
-          this.setState({loading: false});
-          const noValueQuery =
-            values.length === 0 && query.length > 0 ? query : undefined;
-          this.setState({noValueQuery});
-          return values.map(value => {
-            // Wrap in quotes if there is a space
-            const escapedValue =
-              value.indexOf(' ') > -1 ? `"${value.replace('"', '\\"')}"` : value;
-            return {
-              value: escapedValue,
-              desc: escapedValue,
-            };
-          });
-        } else {
-          this.setState({loading: false});
-          return [];
-        }
+      const {location} = this.context.router;
+      const endpointParams = getParams(location.query);
+
+      this.setState({loading: true});
```

Comment:
I actually don't think it would thrash since react only updates state after the macro task completes. But it's in general more understandable I think. 👍